RenderStatePackagerLib/interface/DeviceObjectDescriptorParser.h

```diff
@@ -0,0 +1,305 @@
+    {
+        //TODO
+        return false;
+    }
+#endif
+};
+typedef struct RayTracingGeneralShaderGroupJSON RayTracingGeneralShaderGroupJSON;
+
+
+struct RayTracingTriangleHitShaderGroupJSON 
+{
+    const Char* pClosestHitShader DEFAULT_INITIALIZER(nullptr);
+
+    const Char* pAnyHitShader     DEFAULT_INITIALIZER(nullptr);
+
+#if DILIGENT_CPP_INTERFACE
+    constexpr bool operator == (const RayTracingTriangleHitShaderGroupJSON& RHS) const
+    {
+        //TODO
+        return false;
+    }
+#endif
+};
+typedef struct RayTracingTriangleHitShaderGroupJSON RayTracingTriangleHitShaderGroupJSON;
+
+
+struct RayTracingProceduralHitShaderGroupJSON 
+{
+    const Char* pIntersectionShader DEFAULT_INITIALIZER(nullptr);
+
+    const Char* pClosestHitShader   DEFAULT_INITIALIZER(nullptr);
+
+    const Char* pAnyHitShader       DEFAULT_INITIALIZER(nullptr);
+
+#if DILIGENT_CPP_INTERFACE
+    constexpr bool operator == (const RayTracingProceduralHitShaderGroupJSON& RHS) const
+    {
+        //TODO
+        return false;
+    }
+#endif
+};
+typedef struct RayTracingProceduralHitShaderGroupJSON RayTracingProceduralHitShaderGroupJSON;
+
+struct RayTracingPipelineStateCreateInfoJSON DILIGENT_DERIVE(PipelineStateCreateInfoJSON)
+    const RayTracingGeneralShaderGroupJSON*       pGeneralShaders           DEFAULT_INITIALIZER(nullptr);
+
+    Uint32                                        GeneralShadersCount       DEFAULT_INITIALIZER(0);
+
+    const RayTracingTriangleHitShaderGroupJSON*   pTriangleHitShaders       DEFAULT_INITIALIZER(nullptr);
+
+    Uint32                                        TriangleHitShadersCount   DEFAULT_INITIALIZER(0);
+
+    const RayTracingProceduralHitShaderGroupJSON* pProceduralHitShaders     DEFAULT_INITIALIZER(nullptr);
+
+    Uint32                                        ProceduralHitShadersCount DEFAULT_INITIALIZER(0);
+
+#if DILIGENT_CPP_INTERFACE
+    constexpr bool operator == (const RayTracingPipelineStateCreateInfoJSON& RHS) const 
+    {
+        //TODO
+        return false;
+    }
+#endif
+};
+typedef struct RayTracingProceduralHitShaderGroupJSON RayTracingProceduralHitShaderGroupJSON;
+
+// {355AC9f7-5D9D-423D-AE35-80E0028DE17E}
+static const INTERFACE_ID IID_DeviceObjectDescriptorParser = {0x355AC9F7, 0x5D9D, 0x423D, {0xAE, 0x35, 0x80, 0xE0, 0x02, 0x8D, 0xE1, 0x7E}};
+
+#define DILIGENT_INTERFACE_NAME IDeviceObjectDescriptorParser
+#include "../../../DiligentCore/Primitives/interface/DefineInterfaceHelperMacros.h"
+
+#define DeviceObjectDescriptionParserInclusiveMethods \
+    IObjectInclusiveMethods;                          \
+    IDeviceObjectDescriptorParser DeviceObjectDescriptorParser
+
+
+DILIGENT_BEGIN_INTERFACE(IDeviceObjectDescriptorParser, IObject)
+{
+    VIRTUAL Bool METHOD(GetGraphicsPipelineState)(THIS_
+                                                  const Char* Name,
+                                                  GraphicsPipelineStateCreateInfo REF CreateInfo,
```

Comment:
Here and in other places I'd move GraphicsPipelineStateCreateInfo to Markup struct